src/goImport.ts

```diff
@@ -47,12 +47,34 @@ export function addImport(arg: string) {
 			return vscode.window.activeTextEditor.edit(editBuilder => {
 				editBuilder.insert(new vscode.Position(closeParenLine, 0), '\t"' + imp + '"\n');
 			});
-		} else if (imports.length > 0) {
-			// There are only single import declarations, add after the last one
+		} else if (imports.length > 0 && goConfig.get<boolean>('blockImportsOnly') === false) {
+			// There are only single import declarations, AND the user doesn't want them collapsed.
+			// Add after the last one.
 			let lastSingleImport = imports[imports.length - 1].end;
 			return vscode.window.activeTextEditor.edit(editBuilder => {
 				editBuilder.insert(new vscode.Position(lastSingleImport + 1, 0), 'import "' + imp + '"\n');
 			});
+		} else if (imports.length > 0 && goConfig.get<boolean>('blockImportsOnly') === true) {
+			// There are only single import declarations, but the user wants them collapsed.
+			// NOTE: this most often happens when the go imports tool is run, adds a single import
+			// then later the user manually imports something via this command.
+			let firstImport = imports[0];
+			let lastImport = imports[imports.length - 1];
+
+			// Construct the import block.
+			let importBlock = 'import (\n';
+			imports.forEach(element => {
+				importBlock += '\t"' + element.package + '"\n';
+			});
+			importBlock += '\t"' + imp + '"\n'; // Add the user's new import
+			importBlock += ')\n';
+
+			return vscode.window.activeTextEditor.edit(editBuilder => {
+				// First, delete all the single packages.
+				editBuilder.delete(new vscode.Range(new vscode.Position(firstImport.start, 0), new vscode.Position(lastImport.end, 10000)));
+				// Now, insert the import block.
+				editBuilder.insert(new vscode.Position(pkg.start + 2, 0), importBlock);
```

Comment:
Is anchoring this on `pkg.start` always safe?  What if there are comments in front of the imports or two groups of parenthesized imports?  Can this just be anchored at `firstImport.start` instead?